### DISCUSSION TITLE
Is there a recommended way to implement the search feature?

### DISCUSSION THREAD

**Discussion by EkaLinMan** [TanStack/query#7510] [Q&A] [answered]
﻿Now I have a button to trigger the search action with the input string, the below description and code meet my requirements, but do you have other recommended ways?

- If the input string is different from the previous value, it leads to updating the state and executing the query.
- If the input string is the same as the previous value, l use `refetch` to retrieve the latest data from the server.

```
const [searchString, setSearchString] = useState()

const {data, refetch} = useQuery({
  queryKey: [searchString],
  queryFn: () => fetchItems({searchString}),
})

// We use some way to bind the input element's value as the source of `inputSearchString`.
const handleSearchClick = (inputSearchString) => {
    if (inputSearchString === searchString) {
        refetch();
    } else {
        setSearchString(inputSearchString);
    }
}

<input type="text" />
<button onClick={handleSearchClick})>Search</button>
```

**Comment by TkDodo** [comment, marked as answer]
lgtm 👍
